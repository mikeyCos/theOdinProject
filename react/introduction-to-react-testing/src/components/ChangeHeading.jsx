import { useState } from "react";

export default function ChangeHeading() {
  const [heading, setHeading] = useState("Magnificent Monkeys");
  const clickHandler = () => setHeading("Radical Parrots");

  return (
    <section id="change-heading">
      <h1>{heading}</h1>
      <button type="button" onClick={clickHandler}>
        Change heading
      </button>
    </section>
  );
}
