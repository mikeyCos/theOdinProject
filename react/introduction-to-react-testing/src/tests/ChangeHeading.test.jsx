import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { describe, it, expect } from "vitest";
import ChangeHeading from "../components/ChangeHeading";

describe("ChangeHeading component", () => {
  it("Renders correct ChangeHeading", () => {
    // render(<ChangeHeading />);
    // expect(
    //   screen.getByRole("heading", { name: "Magnificent Monkeys" }).textContent
    // ).toMatch(/magnificent monkeys/i);
    const { container } = render(<ChangeHeading />);
    expect(container).toMatchSnapshot();
  });

  it("Renders radical parrots after button click", async () => {
    const user = userEvent.setup();

    render(<ChangeHeading />);
    const button = screen.getByRole("button", { name: "Change heading" });

    await user.click(button);

    expect(screen.getByRole("heading").textContent).toMatch(/radical parrots/i);
  });
});
