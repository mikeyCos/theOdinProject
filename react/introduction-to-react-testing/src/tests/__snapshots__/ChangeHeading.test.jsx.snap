// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ChangeHeading component > Renders correct ChangeHeading 1`] = `
<div>
  <section
    id="change-heading"
  >
    <h1>
      Magnificent Monkeys
    </h1>
    <button
      type="button"
    >
      Change heading
    </button>
  </section>
</div>
`;

exports[`ChangeHeading component > renders correct ChangeHeading 1`] = `
<div>
  <section
    id="change-heading"
  >
    <h1>
      Magnificent Monkeys
    </h1>
    <button
      type="button"
    >
      Change heading
    </button>
  </section>
</div>
`;
